### PR TITLE
Recipe create/edit dialog

### PR DESCRIPTION
**Pull request by marcin-hoa**

### CODE REVIEW

File: libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts [hunk 1]
```diff
@@ -28,5 +29,59 @@ const loadAll$ = createEffect(
...
+  (actions$ = inject(Actions)) => {
+    const repo = inject(RecipesRepository);
+    const dialog = inject(MatDialog);
+
+    return actions$.pipe(
+      ofType(RecipesUiActions.createRecipe),
+      exhaustMap((action) => {
+        return repo.create(action.recipeDto).pipe(
+          map((res) => {
+            dialog.closeAll();
```
Comment (DDonochVA) [inline libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts]: it shouldn't be here; dialog is an implementation detail, so we'd rather not add any unrelated dependencies. Imagine needing to move your forms from a dialogue to regular components. That would necessitate adjustments to the effects logic, an unnecessary complication if we maintain a streamlined and independent architecture.

File: libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts [hunk 2]
```diff
@@ -28,5 +29,59 @@ const loadAll$ = createEffect(
...
+      }),
+    );
+  },
+  { functional: true },
+);
+
+const editRecipe$ = createEffect(
+  (actions$ = inject(Actions)) => {
+    const repo = inject(RecipesRepository);
+    const dialog = inject(MatDialog);
```
Comment (DDonochVA) [inline libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts]: as above

File: libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts (hunk 1 above)
Comment (marcin-hoa) [reply libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts]: @DDonochVA Fixed. I moved to the `feature-recipe-edit-dialog.component.ts` itself. It listens to `RecipesApiActions.createRecipeSuccess` and `RecipesApiActions.editRecipeSuccess` actions.

File: libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts (hunk 2 above)
Comment (marcin-hoa) [reply libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts]: as above :)

**Review by DDonochVA — APPROVED** [review]

**Pull request merged**